### PR TITLE
Added option to show settings.

### PR DESCRIPTION
**Pull request by garyjN7**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| Breaking change?  | no
| New feature?      | yes #2960
| Deprecations?     | no
| Spec compliancy?  | no
| Tests added/pass? | no
| Fixed tickets     | #2960
| License           | MIT
| Doc PR            | 

Added settings option to babel-cli: --settings or -S.  Can be used in conjunction with zero or more files e.g.

## EDIT by hzoo

```
babel --settings
babel script1.js script2.js --settings
```

Output will show node version, npm version, package versions, and general settings and specific settings per file.   Example output shown below...

```
node version: v6.5.0
npm version: 3.10.3
packages:
babel-test-proj@1.0.0 C:\Dev\mytest
`-- babel-cli@6.18.0
  +-- babel-core@6.18.0
  | +-- babel-code-frame@6.16.0
  | +-- babel-generator@6.18.0
  | +-- babel-helpers@6.16.0
  | +-- babel-messages@6.8.0
  | +-- babel-template@6.16.0
  | +-- babel-traverse@6.18.0
  | +-- babel-types@6.18.0
  | +-- babylon@6.13.0
  +-- babel-polyfill@6.16.0
  +-- babel-register@6.18.0
  +-- babel-runtime@6.18.0

--- General C:\Dev\mytest options ---
filename :  unknown
filenameRelative :  unknown
inputSourceMap :  undefined
env :  {}
mode :  undefined
retainLines :  false
highlightCode :  true
suppressDeprecationMessages :  false
presets :  []
plugins :  []
ignore :  [ /(?:(?=.)foo\.js)/i,
  /(?:(?=.)bar\/(?:(?!(?:\/|^)\.).)*?\/(?!\.)(?=.)[^\/]*?\.js)/i ]
only :  undefined
code :  true
metadata :  true
ast :  true
extends :  undefined
comments :  true
shouldPrintComment :  undefined
wrapPluginVisitorMethod :  undefined
compact :  auto
minified :  false
sourceMap :  undefined
sourceMaps :  undefined
sourceMapTarget :  unknown
sourceFileName :  unknown
sourceRoot :  undefined
babelrc :  true
sourceType :  module
auxiliaryCommentBefore :  undefined
auxiliaryCommentAfter :  undefined
resolveModuleSource :  undefined
getModuleId :  undefined
moduleRoot :  undefined
moduleIds :  false
moduleId :  undefined
passPerPreset :  false
parserOpts :  false
generatorOpts :  false
basename :  unknown

--- script1.js options ---
filename :  script1.js
filenameRelative :  script1.js
sourceMapTarget :  script1.js
sourceFileName :  script1.js
basename :  script1

--- script2.js options ---
filename :  script2.js
filenameRelative :  script2.js
ignore :  []
compact :  true
sourceMapTarget :  script2.js
sourceFileName :  script2.js
basename :  script2
```

### CODE REVIEW

File: packages/babel-cli/src/babel/util.js [hunk 1]
```diff
@@ -81,3 +83,48 @@ export function requireChokidar() {
...
+    });
+    console.log();
+  };
+
+  process.stdout.write(`node version: `);
+  child.execSync("node -v", {stdio:[0, 1]});
+  process.stdout.write(`npm version: `);
+  child.execSync("npm -v", {stdio:[0, 1]});
+  process.stdout.write(`packages:\n`);
+  child.execSync("npm list", {stdio:[0, 1]});
```
Comment (hzoo) [inline packages/babel-cli/src/babel/util.js]: might be useful to filter out non-babel packages for `npm list` -> `npm list | grep babel` but can't assume `grep` is available

Comment (garyjN7) [inline packages/babel-cli/src/babel/util.js]: No problem.  I have pushed another commit that will just print babel package versions.

File: packages/babel-cli/src/babel/util.js [hunk 2]
```diff
@@ -6,6 +7,7 @@ let util      = require("babel-core").util;
 let path      = require("path");
 let fs        = require("fs");
 let _         = require("lodash");
+let each      = require("lodash/each");
```
Comment (abouthiroppy) [inline packages/babel-cli/src/babel/util.js]: Is this necessary? 
Lodash was called at line 9 so you should write `_.each` or `let each = _.each`

Comment (garyjN7) [inline packages/babel-cli/src/babel/util.js]: It was to keep it consistent with other modules where each has been used like this, but I have changed it as per your suggestion.

**Comment by JLHwung** [comment]
Since Babel 7.11, you can now print babel configs via `BABEL_SHOW_CONFIG_FOR`: https://babel.dev/docs/en/configuration#print-effective-configs

**Pull request closed without merging**